app/Project.php

```diff
@@ -145,4 +145,84 @@ public function user()
     {
         return $this->belongsTo(User::class);
     }
+
+    /**
+     *
+     *
+     * @return 
+     */
+    public function followers()
+    {
+        return $this->hasMany(ProjectFollower::class);
+    }
+
+    /**
+     * Get the count of followers for the project
+     *
+     * @return
+     */
+    public function getFollowerCount()
+    {
+        return count(ProjectFollower::where('project_id', '=', $this->id)->get());
+    }
+
+    /**
+     * Check if user is a follower of this project
+     *
+     * @return
+     */
+    public function isUserFollower()
+    {
+        if (!Auth::guest()) {
+            $isUserFollower = ProjectFollower::where('user_id', '=', Auth::user()->id)->where('project_id', '=', $this->id)->first();
+            if ($isUserFollower != null)
+                $isUserFollower = true;
+            else
+                $isUserFollower = false;
+        }
+        else
+            $isUserFollower = false;
+
+        return $isUserFollower;
+    }
+
+    /**
+     * Adds a follower to the project
+     *
+     * @return
+     */
+    public function addFollower()
+    {
+        $follower = ProjectFollower::create([
+            'user_id' => Auth::user()->id,
+            'project_id' => $this->id
+        ]);
+
+        return $this->followers()->save($follower);
+    }
+
+    /**
+     * Deletes a follower to the project
+     *
+     * @return
+     */
+    public function deleteFollower()
+    {
+        foreach ($this->followers as $follower)
+        {
+            if ($follower->user_id == Auth::user()->id)
+                $follower->delete();
+        }
+    }
+
+    public function getFollowerIDs()
+    {
+        $followerArray = [];
+        foreach ($this->followers as $follower)
```

Comment:
Interesting didn't know about that, but I ended up deleting the function because it actually wasn't being used.